### PR TITLE
Remove Reference to Common ContractReader

### PR DESCRIPTION
**Pull request by EasterTheBunny**
The interface `ContractReader` in common is expected to be implemented by numerous relay/chains and will have strong type requirements to ensure no introduction of breaking changes. Using a local interface in this repo also shields ccip from further breaking changes and simplifies mocking.

The common interface `ContractReader` will not be mockable by mockery with the above mentioned expectations on relay/chain implementations.

### CODE REVIEW

File: execute/factory.go [hunk 1]
```diff
@@ -56,7 +57,7 @@ type PluginFactory struct {
 	homeChainReader   reader.HomeChain
 	estimateProvider  gas.EstimateProvider
 	tokenDataObserver tokendata.TokenDataObserver
-	contractReaders   map[cciptypes.ChainSelector]types.ContractReader
+	contractReaders   map[cciptypes.ChainSelector]contractreader.Reader
```
Comment (winder) [inline execute/factory.go:60]: I believe this would require coordinated changes in `chainlink` where this function is used

**Comment by makramkd** [comment]
If we want to do this lets re-open #125 cc @winder

File: execute/factory.go (hunk 1 above)
Comment (makramkd) [reply execute/factory.go:60]: Yeah preferably coordination not needed IMO

Comment (EasterTheBunny) [reply execute/factory.go:60]: Can `chainlink` not pass `map[cciptypes.ChainSelector]types.ContractReader` to `NewPluginFactory`? Not sure if this works or not.

**Comment by EasterTheBunny** [comment]
> If we want to do this lets re-open #125 cc @winder

I am offering this as a proposed solution on handling breaking changes going forward. Other solutions welcome.

File: execute/factory.go (hunk 1 above)
Comment (winder) [reply execute/factory.go:60]: I'm pretty sure map values work a little different and you have to convert it manually
```
readers := make(map[cciptypes.ChainSelector]reader.ContractReaderFacade)
	for chain, cr := range p.contractReaders {
		readers[chain] = cr
	}
```

You can see the error by leaving the `types.ContractReader` type on `func NewPluginFactory(...)`

Comment (EasterTheBunny) [reply execute/factory.go:60]: This doesn't show any errors

```
type PluginFactory struct {
    contractReaders map[cciptypes.ChainSelector]contractreader.Reader
    // ... other stuff
}

func NewPluginFactory(
	lggr logger.Logger,
	ocrConfig reader.OCR3ConfigWithMeta,
	commitCodec cciptypes.CommitPluginCodec,
	msgHasher cciptypes.MessageHasher,
	homeChainReader reader.HomeChain,
	contractReaders map[cciptypes.ChainSelector]types.ContractReader,
	chainWriters map[cciptypes.ChainSelector]types.ChainWriter,
) *PluginFactory {
	readers := make(map[cciptypes.ChainSelector]contractreader.Reader)
	for chain, cr := range contractReaders {
		readers[chain] = cr
	}

	return &PluginFactory{
		lggr:            lggr,
		ocrConfig:       ocrConfig,
		commitCodec:     commitCodec,
		msgHasher:       msgHasher,
		homeChainReader: homeChainReader,
		contractReaders: readers,
		chainWriters:    chainWriters,
	}
}
```

Comment (winder) [reply execute/factory.go:60]: Right, you needed to map the types manually. The original changes would have required that mapping in `chainlink` code

**Review by jmank88 — APPROVED** [review]

**Comment by asoliman92** [comment]
Closing as @winder already created a Reader Facade

**Pull request closed without merging**